tree/ntuple/v7/src/RPageStorage.cxx

```diff
@@ -1014,7 +1014,7 @@ void ROOT::Experimental::Internal::RPagePersistentSink::CommitSealedPageV(
       locatorIndexes.reserve(locatorIndexes.size() + rangeSize);
 
       for (auto sealedPageIt = range.fFirst; sealedPageIt != range.fLast; ++sealedPageIt) {
-         if (!fFeatures.fCanMergePages || !sealedPageIt->GetHasChecksum()) {
+         if (!fFeatures.fCanMergePages || !fOptions->GetEnableSamePageMerging() || !sealedPageIt->GetHasChecksum()) {
```

Comment:
Should we still check for the checksum here and not only for same page merging?

Reply:
Yes right; I turned it into an explicit assertion that is checked before unwrapping the checksum.